### ISSUE TITLE
Forbidden in 3.4

### ISSUE THREAD

**Issue by shacker** [mingchen/django-cas-ng#17]
I have a super simple django-cas-ng test project with nothing installed but Django 1.7 and  django-cas-ng, with these settings:

```
CAS_SERVER_URL = 'https://cas.oursite.edu/cas/login'
CAS_ADMIN_PREFIX = '/admin'
CAS_LOGOUT_COMPLETELY = True
```

plus the login/logout URLs shown in the docs. 

With versions 3.1, 3.2 and 3.3, it works just fine. But when I upgrade to 3.4 or 3.4.1 I get:

http://127.0.0.1:8000/accounts/login?next=%2F&ticket=ST-2138-43ZolaFcMAeLcZjAK-cas.oursite.edu

```
    Forbidden
    Login failed.
```

(403 on the GET request as shown in runserver). Is there an additional setting or configuration I need to use when upgrading?

**Comment by jpg18** [comment]
@shacker, the issue you've described sounds exactly like an issue I encountered when I upgraded django-cas-ng to versions >= 3.4.0.

I searched through the commits until I found the one between v3.3.0 and v.3.40 that first causes the issue on my system.  This was commit 96c67bf459977b582d0d3dd025014b904d2a63ae.

I then applied portions of that commit to its parent until I found the minimum changes that caused the issue.  I've created a patch (5eb8cc37fb94f1a0dce4045299a778c77861616e) that allows me to use the latest version of django-cas-ng.

The issue was caused by a bad service URL, which appears in the redirect URL when logging.  In my case, the service URL contained a duplicated query string, such as

    http://host/accounts/login/?next=/path/to/app/&next=/path/to/app/

Here I've removed all the HTML character escaping for clarity. The second "next" was double-escaped.

I've noticed that the URL you shared in your original post doesn't have this duplication, but that may be because my problem appeared in the address bar before entering my login credentials at my university's Single Sign-On interface, i.e., before I am given a ticket.  Your URL was clearly constructed after you were given a ticket. So, this patch may work for you too.
